fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseSpecificSQLGenerator.java

```diff
@@ -161,7 +161,7 @@ public String dateDiff(String date1, String date2) {
 
     public String castChar(String sql) {
         if (databaseTypeResolver.isMySQL()) {
-            return format("CAST(%s AS CHAR)", sql);
+            return format("CAST(%s AS CHAR) COLLATE utf8mb4_unicode_ci", sql);
```

Comment:
@IOhacker Can you help me understand this more? By default i would not be big fan of hardcoding any collation... Can you please share here and or in the story a little bit more information about this bug?